### DISCUSSION TITLE
Grouping separator is not taken from Windows regional settings

### DISCUSSION THREAD

**Discussion by GeriZ** [dbeaver/dbeaver#34351] [Q&A]
### Description

I have set the decimal point in the Windows/Regional settings(Germany) as thousands grouping separator, so expecting numbers to be formatted like that: 123.456.789,00. But this settings is ignored when presenting data in a result set. "Use grouping" is active and "Grouping size" is 3 in the Dbeaver preferences

PS: I wanted to add a screenshot but regardless if I paste or drop or add the screenshot file it does not appear. I've tried it with  Firefox and Chrome but it did'nt work in both

### DBeaver Version

Community Edition 24.1

### Operating System

Windows Server 2016

### Database and driver

DB2 11.5.9

**Comment by E1izabeth** [comment]
Have you tried changing this settings?
![image](https://github.com/dbeaver/dbeaver/assets/28875055/590f5cdc-f8ef-4914-819d-118dbec132b0)

**Comment by GeriZ** [comment]
Interesting! When I use  the language "en - English" instead of "de - German" and country "AT - Austria" the sample dialog suddenly shows the correct value 1.234.567.890,012

But when I use language "de - German" and country "AT - Austria" the sample dialog shows the incorrect value 1 234 567 890,012

Is this intended? Because german speaking countries usually use the decimal point as separation character(and the comma before the decimals). I would not expect blanks when using "de - German" as language

**Comment by ShadelessFox** [comment]
DBeaver doesn't use system-wide formatting settings - it's a limitation of the platform on which DBeaver is built (in our case, the Java programming language). Instead, formatting settings are integrated with the platform, following a specific ISO standard IIRC. So, to me, this behavior is expected.

**Reply by GeriZ** [reply]
I understand but it's still unclear for me why "en - English" shows the european notation 1.234.567.890,012 and not 1,234,567,890.012. And why "de - German" shows blanks as seperators -> 1 234 567 890,012
